### DISCUSSION TITLE
Enable SSO and authorization option not found for either SSH keys and Personal Access Token

### DISCUSSION THREAD

**Discussion by adhadse** [community/community#23864] [New to GitHub]
<p>Hello Community, I am trying to build Hugo Website and deploy using Github Actions to push to another repo under my name with <code>github.io</code>. the problem is that even upon adding the Personal access token to repo’s secrets, I am not able to push to the external repo, as the the error says,</p>
<p><code>Action failed with "The generated GITHUB_TOKEN (github_token) does not support to push to an external repository. Use deploy_key or personal_token. "</code>.</p>
<p>I stumbled upon this <a href="https://stackoverflow.com/a/65091631/8277795" rel="noopener nofollow ugc">Stackoverflow answer</a>, but to my notice, option to enable and authorize SSO to PAT seems to be missing.</p>
<p>Any help possible?</p>

**Comment by byrneh** [comment]
<p>HI <a class="mention" href="/adhadse">@adhadse</a>, welcome</p>
<p>Authorizing a personal access token for use with SAML single sign-on is documented here:<br>
<a href="https://docs.github.com/en/github/authenticating-to-github/authorizing-a-personal-access-token-for-use-with-saml-single-sign-on" rel="noopener nofollow ugc">authorizing-a-personal-access-token-for-use-with-saml-single-sign-on</a>.</p>
<p>HOWEVER this is only relevant if your Organization is configured for SSO to an IdP, which is not an available feature for user repositories or free plan Organizations.  You will know if SAML is enabled for your organization as you have to provide IdP credentials in addition to your GitHub login credentials when accessing any organization repositories.</p>
<p>I am not overly familiar with actions, but the message you pasted states ‘The generated GITHUB_TOKEN (github_token) does not support to push to an external repository’.  This implies the personal access token you created is not being used or configured correctly in your action code.</p>
<p>You can probably look around and find an example that you could follow to correct this problem, or another helpful community member may post a link.<br>
I think you will need to specify the created personal access token secrets  in the push to the external repository and not the default GITHUB_TON created by the workflow action, that does not have permission on external repositories</p>

**Comment by adhadse** [comment, marked as answer]
<p>Well thanks for the your valuable reply and concern. My problem got fixed, I was typing <code>github_token: ${{ secrets.token )) instead of </code>personal_token: ${{ secrets.token }}. Should Have read the docs prior to this.</p>

**Comment by byrneh** [comment]
<p><a class="mention" href="/adhadse">@adhadse</a> great news :slight_smile:<br>
if you mark the post as solution, it will be clearer for anyone looking searching for the same problem</p>
<p>Good luck with your website and action deployments.</p>

**Comment by ygormutti** [comment]
<p>So, does it mean it’s impossible to use the automatically generated GITHUB_TOKEN secret in orgs where SAML SSO is enabled?</p>
